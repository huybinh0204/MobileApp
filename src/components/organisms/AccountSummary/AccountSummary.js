import { useNavigation } from '@react-navigation/core';
import React, { useMemo } from 'react';
import { useSelector } from 'react-redux';
import { Balance } from '_components/atoms';
import { ICONS, NAVIGATION } from '_constants';
import strings from '_localization';
import { AccountSelectors } from '_store/account';
import { CustomerSelectors } from '_store/customer';
import { getTimeBlock } from '_utilities/date';
import { normalize } from '_utilities/screen';
import {
  BankAccountLabel,
  Column,
  Container,
  Content,
  Greeting,
  Illustration,
  UserName,
  ViewAccountButton,
} from './AccountSummary.styles';

const AccountSummary = () => {
  const { navigate } = useNavigation();

  const account = useSelector(AccountSelectors.getAccountInfo);
  const customer = useSelector(CustomerSelectors.getCustomer);
  const dayTime = useMemo(() => getTimeBlock(), []);

  const navigateToAccount = () => {
    navigate(NAVIGATION.accounts.stack, { screen: NAVIGATION.accounts.main });
  };

  return (
    <Container testID="AccountSummary">
      <Greeting>
        {strings.home.greeting[dayTime]} <UserName>{customer?.firstName}</UserName>
      </Greeting>
      <Content>
        <Column>
          <BankAccountLabel>{strings.home.accountBalance}</BankAccountLabel>
          <Balance
            testID="AccountSummaryBalance"
            amount={account?.balance}
            typography="heading3"
            placeholder={`${strings.currencySymbol}--.--`}
          />
          <ViewAccountButton onPress={navigateToAccount}>
            {strings.home.viewAccountButton}
          </ViewAccountButton>
        </Column>
        <Column>
          <Illustration
            icon={ICONS.accountSummary}
            height={normalize(125)}
            width={normalize(150)}
          />
        </Column>
      </Content>
    </Container>
  );
};

export default AccountSummary;
