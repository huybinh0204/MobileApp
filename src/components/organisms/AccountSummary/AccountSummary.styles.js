import styled from '@emotion/native';
import { BaseText, IconSvg, MainButton } from '_components/atoms';

export const Container = styled.View(({ theme }) => ({
  marginBottom: theme.spacing.s,
  paddingHorizontal: theme.spacing.s,
  paddingBottom: theme.spacing.s,
}));

export const Greeting = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading5,
  color: theme.colors.beta900,
  fontWeight: 'normal',
  marginBottom: theme.spacing.m,
}));

export const UserName = styled(BaseText)({
  fontWeight: 'bold',
});

export const Content = styled.View({
  flexDirection: 'row',
});

export const Column = styled.View({
  flex: 1,
  justifyContent: 'flex-end',
});

export const Illustration = styled(IconSvg)({
  alignSelf: 'flex-end',
});

export const BankAccountLabel = styled(BaseText)(({ theme }) => ({
  ...theme.typography.tinyContentRegular,
  color: theme.colors.beta500,
}));

export const ViewAccountButton = styled(MainButton)(({ theme }) => ({
  marginTop: theme.spacing.s,
}));
