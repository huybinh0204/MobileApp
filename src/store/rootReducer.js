import { combineReducers } from 'redux';
import actionWatchReducer from 'redux-action-watch/lib/reducer';
import { resettableReducer } from 'reduxsauce';
import { AccountReducer } from './account';
import { AuthenticationReducer } from './authentication';
import { CustomerReducer } from './customer';
import { FundDirectDepositStoreReducer } from './pageStore/fundDirectDepositStore';
import { RegisterReducer } from './register';
import { TransferReducer } from './transfer';

const resettable = resettableReducer('RESET');

export default combineReducers({
  account: resettable(AccountReducer),
  auth: resettable(AuthenticationReducer),
  customer: resettable(CustomerReducer),
  fundDirectDepositStoreReducer: resettable(FundDirectDepositStoreReducer),
  register: RegisterReducer,
  transfer: resettable(TransferReducer),
  watcher: actionWatchReducer, // Must be last
});
