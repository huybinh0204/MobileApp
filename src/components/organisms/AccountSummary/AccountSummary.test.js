import { useNavigation } from '@react-navigation/native';
import { render } from '@testing-library/react-native';
import React from 'react';
import { createStore } from 'redux';
import rootReducer from '_store/rootReducer';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import { navigationMock } from '_test/mocks/navigation';
import AccountSummary from './AccountSummary';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn(),
}));

describe('AccountSummary', () => {
  let mockStore;
  const mockedNavigation = navigationMock(jest.fn);

  beforeEach(() => {
    useNavigation.mockReturnValue(mockedNavigation);
    mockStore = createStore(rootReducer, {
      account: {
        accountInfo: {
          balance: 1000.55,
        },
      },
    });
  });

  it('should match the snapshot', () => {
    const { toJSON } = render(
      <ComponentWithProviders store={mockStore} component={AccountSummary} />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should show the balance formatted', () => {
    const { getByText } = render(
      <ComponentWithProviders store={mockStore} component={AccountSummary} />
    );

    const balance = getByText('$1,000.55');

    expect(balance).toBeTruthy();
  });
});
