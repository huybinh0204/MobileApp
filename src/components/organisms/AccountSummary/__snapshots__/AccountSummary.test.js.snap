// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountSummary should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "marginBottom": 29,
                              "paddingBottom": 29,
                              "paddingHorizontal": 29,
                            },
                            undefined,
                          ]
                        }
                        testID="AccountSummary"
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": "#1F1D25",
                                "fontFamily": "Poppins-SemiBold",
                                "fontSize": 36,
                                "fontWeight": "normal",
                                "lineHeight": 50.5,
                                "marginBottom": 58,
                              },
                              undefined,
                            ]
                          }
                        >
                          Good evening,
                           
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "fontWeight": "bold",
                                },
                                undefined,
                              ]
                            }
                          />
                        </Text>
                        <View
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "flex": 1,
                                  "justifyContent": "flex-end",
                                },
                                undefined,
                              ]
                            }
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#5F6A77",
                                    "fontFamily": "Poppins",
                                    "fontSize": 21.5,
                                    "lineHeight": 30.5,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Account Balance
                            </Text>
                            <Text
                              allowFontScaling={false}
                              isNegative={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#1F1D25",
                                    "fontFamily": "Poppins-SemiBold",
                                    "fontSize": 50.5,
                                    "lineHeight": 70.5,
                                  },
                                  undefined,
                                ]
                              }
                              typography="heading3"
                            >
                              $1,000.55
                            </Text>
                            <View
                              accessibilityState={
                                Object {
                                  "disabled": false,
                                }
                              }
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "alignSelf": "stretch",
                                  "backgroundColor": "#3D3397",
                                  "borderRadius": 10,
                                  "marginTop": 29,
                                  "opacity": 1,
                                  "paddingHorizontal": 45.5,
                                  "paddingVertical": 31,
                                }
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                disabled={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontFamily": "Poppins-SemiBold",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                      "textAlign": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                View account
                              </Text>
                            </View>
                          </View>
                          <View
                            style={
                              Array [
                                Object {
                                  "flex": 1,
                                  "justifyContent": "flex-end",
                                },
                                undefined,
                              ]
                            }
                          >
                            <SvgMock
                              height={226.5}
                              style={
                                Array [
                                  Object {
                                    "alignSelf": "flex-end",
                                  },
                                  undefined,
                                ]
                              }
                              width={271.5}
                            />
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
